gns3server/handlers/api/file_handler.py

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright (C) 2015 GNS3 Technologies Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import asyncio
+
+from ...web.route import Route
+from ...schemas.file import FILE_STREAM_SCHEMA
+
+
+class FileHandler:
+
+    @classmethod
+    @Route.get(
+        r"/files/stream",
+        description="Stream a file from the server",
+        status_codes={
+            200: "File retrieved",
+            404: "File doesn't exist"
+        },
+        input=FILE_STREAM_SCHEMA
+    )
+    def read(request, response):
+        try:
+            with open(request.json.get("location"), "rb") as f:
+                loop = asyncio.get_event_loop()
+                response.content_type = "application/octet-stream"
+                response.set_status(200)
+                response.enable_chunked_encoding()
+                # Very important: do not send a content lenght otherwise QT close the connection but curl can consume the Feed
+                response.content_length = None
+
+                response.start(request)
+
+                while True:
+                    data = yield from loop.run_in_executor(None, f.read, 16)
```

Comment:
I have only little concern about using executors, what happen when we hit max_workers limit which I think is 4 by default? According to docs a task waits for an executor to become available, this probably means that we can have a maximum of 4 concurrent captures.